# Variable Notes

1. pclass: A proxy for socio-economic status (SES)
    1st = Upper
    2nd = Middle
    3rd = Lower

2. age: Age is fractional if less than 1. If the age is estimated, is it in the form of xx.5

3. sibsp: The dataset defines family relations in this way...
4. Sibling = brother, sister, stepbrother, stepsister
5. Spouse = husband, wife (mistresses and fiancés were ignored)

6. parch: The dataset defines family relations in this way...
7. Parent = mother, father
8. Child = daughter, son, stepdaughter, stepson
Some children travelled only with a nanny, therefore parch=0 for them

# Data dictionary
1. "survival	Survival	0 = No, 1 = Yes",
2. "pclass	Ticket class	1 = 1st, 2 = 2nd, 3 = 3rd",
3. "sex	Sex	",
4. "Age	Age in years	",
5. "sibsp	# of siblings / spouses aboard the Titanic	",
6. "parch	# of parents / children aboard the Titanic	",
7. "ticket	Ticket number	",
8. "fare	Passenger fare	",
9. "cabin	Cabin number	",
10. "embarked	Port of Embarkation	C = Cherbourg, Q = Queenstown, S = Southampton",

In [1]:
data_dictionary = [
"survival Survival 0 = No, 1 = Yes",
"pclass Ticket class 1 = 1st, 2 = 2nd, 3 = 3rd",
"sex Sex ",
"Age Age in years ",
"sibsp # of siblings / spouses aboard the Titanic ",
"parch # of parents / children aboard the Titanic ",
"ticket Ticket number ",
"fare Passenger fare ",
"cabin Cabin number ",
"embarked Port of Embarkation C = Cherbourg, Q = Queenstown, S = Southampton",
    ]

In [2]:
data_dictionary

['survival Survival 0 = No, 1 = Yes',
 'pclass Ticket class 1 = 1st, 2 = 2nd, 3 = 3rd',
 'sex Sex ',
 'Age Age in years ',
 'sibsp # of siblings / spouses aboard the Titanic ',
 'parch # of parents / children aboard the Titanic ',
 'ticket Ticket number ',
 'fare Passenger fare ',
 'cabin Cabin number ',
 'embarked Port of Embarkation C = Cherbourg, Q = Queenstown, S = Southampton']

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
# ttrain - titanic train data
ttrain = pd.read_csv('./titanic/train.csv')
ttrain.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
ttrain.shape

(891, 12)

In [6]:
# filter how many survived


### Now lets split out the surname and name from Name column

In [7]:
ttrain.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [8]:
ttrain.Age.max()

80.0

In [9]:
ttrain.Age.min()

0.42

In [10]:
ttrain.Age.duplicated().value_counts()

True     802
False     89
Name: Age, dtype: int64

In [11]:
ttrain.Age.value_counts()

24.00    30
22.00    27
18.00    26
19.00    25
30.00    25
         ..
55.50     1
70.50     1
66.00     1
23.50     1
0.42      1
Name: Age, Length: 88, dtype: int64

In [12]:
ttrain.Survived[ttrain.Survived == 1].count().sum()

342

In [13]:
891-342

549

In [14]:
ttrain.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [15]:
ttrain.Age.mean()

29.69911764705882

In [16]:
ttrain.Age.var()

211.01912474630802

In [17]:
ttrain.Age.std()

14.526497332334042

* we the help statistical operations we oberved that it is some how appropriate to fill the null age values with its mean
* But this thing is not ideal may be the missing data Age far away from the mean so, in that case it leads to false pardications

In [18]:
# Fill missing age values with the mean of the Age
ttrain.Age = ttrain.Age.fillna(ttrain.Age.mean())
ttrain.Age.isna().sum()

0

In [19]:
ttrain.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

* since in cabin 687 data is missing
* therefore it is better drop the cabin column

In [20]:
ttrain.drop('Cabin',axis=1)

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex        Age  \
0                              Braund, Mr. Owen Harris    male  22.000000   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.000000   
2                               Heikkinen, Miss. Laina  female  26.000000   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.000000   
4                             Allen, Mr. William Henry    male  35.000000   
..                                                 ...     ...        ...   
886                              Montvila, Rev. Juozas    male  27.000000   
887                       Graham, Miss. Margaret Edith  female  19.000000   
888           Johnston, Miss. Catherine Helen "Carrie"  female  29.699118   
889                              Behr, Mr. Karl Howell    male  26.000000   
890                                Dooley, Mr. Patrick    male  32.000000   

     SibSp  Parch            Ticket     Fare Embarked  
0        1      0         A/5 21171   7.2500        S  
1        1      0          PC 17599  71.2833        C  
2        0      0  STON/O2. 3101282   7.9250        S  
3        1      0            113803  53.1000        S  
4        0      0            373450   8.0500        S  
..     ...    ...               ...      ...      ...  
886      0      0            211536  13.0000        S  
887      0      0            112053  30.0000        S  
888      1      2        W./C. 6607  23.4500        S  
889      0      0            111369  30.0000        C  
890      0      0            370376   7.7500        Q  

[891 rows x 11 columns]